fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJournalCursor.java

```diff
@@ -79,54 +78,109 @@ private BDBJournalCursor(BDBEnvironment env, long fromKey, long toKey) throws Ex
                 break;
             }
         }
+        nextDbPositionIndex -= 1;
 
         if (dbName == null) {
-            LOG.error("Can not find the key:{}, fail to get journal cursor. will exit.", fromKey);
-            System.exit(-1);
+            throw new JournalException(String.format("Can not find the key:%d, fail to get journal cursor!", fromKey));
+        }
+    }
+
+    private boolean shouldOpenDatabase() {
+        // the very first time
+        if (database == null) {
+            return true;
+        }
+        // if current db does not contain any more data, then we go to search the next db
+        return nextDbPositionIndex < dbNames.size() && currentKey == dbNames.get(nextDbPositionIndex);
+    }
+
+    protected void openDatabaseIfNecessary() throws InterruptedException, JournalException {
+        if (!shouldOpenDatabase()) {
+            return;
+        }
+
+        String dbName = Long.toString(dbNames.get(nextDbPositionIndex));
+        JournalException exception = null;
+        for (int i = 0; i != RETRY_TIME; ++ i) {
```

Comment:
I think it is safer to use i < RETRY_TIME

Reply:
Done